app/scripts/lib/stream-utils.js

```diff
@@ -24,3 +24,15 @@ export function setupMultiplex(connectionStream) {
   });
   return mux;
 }
+
+/**
+ * Checks if a stream is usable
+ *
+ * @param {stream.Stream} stream - the stream to check
+ * @returns {boolean} if the stream can be written to
+ */
+export function isStreamWritable(stream) {
+  return Boolean(
+    stream.writable && !stream.destroyed && !stream._writableState?.ended,
```

Comment:
can you add comments referencing these fields / what stream version they apply to

Reply:
- `stream.writable`:
  - [`readable-stream-3`](https://nodejs.org/docs/latest-v10.x/api/stream.html#stream_writable_writable) (confusingly: not mentioned in docs for streamsv2 and not consistently implemented there, despite v3 docs mentioning it as older)
  - [`readable-stream-4`/`NodeStream`](https://nodejs.org/docs/latest-v18.x/api/stream.html#writablewritable) (here it's mentioned as introduced much later)
- `stream.destroyed`:
  - [`readable-stream-4`/`NodeStream`](https://nodejs.org/docs/latest-v18.x/api/stream.html#writablewritable) (mentions it as introduced in v2 despite being absent from both implementation and docs of v2 and v3)
- `stream._writableState.ended`:
  - Present in all implementations, seems like the most reasonable fallback for legacy.

Reply:
I think he's implying that we ought to add inline comments for this, which I agree is a good idea.

Reply:
Added a comment, hopefully it won't hurt more than it helps..